pangeo_forge_recipes/patterns.py

```diff
@@ -248,6 +248,12 @@ def items(self):
         for key in self:
             yield key, self[key]
 
+    def sha256(self):
+        """Compute a sha256 hash for the instance."""
+        from .serialization import pattern_blockchain
```

Comment:
Why do the import here? Isn't it better to put it at the top of the module?

Reply:
idk! 🙃 https://github.com/pangeo-forge/pangeo-forge-recipes/pull/349#issuecomment-1111275762

I'll move it

Reply:
hmm actually if it's not within the function it creates a circular import, because in `serialization` we import from `.patterns` to get `FilePattern` and `Index` for type hinting

```python-traceback
ImportError while loading conftest '/Users/charlesstern/Dropbox/pangeo/pangeo-forge-recipes/tests/conftest.py'.
tests/conftest.py:34: in <module>
    from pangeo_forge_recipes.patterns import (
pangeo_forge_recipes/patterns.py:23: in <module>
    from .serialization import pattern_blockchain
pangeo_forge_recipes/serialization.py:8: in <module>
    from .patterns import FilePattern, Index
E   ImportError: cannot import name 'FilePattern' from partially initialized module 'pangeo_forge_recipes.patterns' (most likely due to a circular import) (/Users/charlesstern/Dropbox/pangeo/pangeo-forge-recipes/pangeo_forge_recipes/patterns.py)
```

Reply:
Hmmm we should really avoid circular dependency between modules. That means you should probably move the main pattern_blockchain into patterns.py. And the same for any anything else that imports from other modules. We can still leave the helper functions in serialization.py.

Reply:
ok I think this is fixed in the latest commits